### PR TITLE
Add support for client key password for HTTPS connections (IDFGH-5700)

### PR DESCRIPTION
**Pull request by joncmaloney**
I've tested password protected PEM file format keys that use RSA. I've also tested the inverse case using a non password protected key and this still works. Provided the element client_key_password is null or client_key_password_len is 0.

### CODE REVIEW

**Comment by joncmaloney** [comment]
resolves #7418

**Comment by Alvin1Zhang** [comment]
Thanks for your contribution.

File: components/esp_http_client/esp_http_client.c [hunk 1]
```diff
@@ -632,6 +632,12 @@ esp_http_client_handle_t esp_http_client_init(const esp_http_client_config_t *co
         }
     }
 
+    if (config->client_key_password) {
+        if(config->client_key_password_len > 0){
```
Comment (AshUK) [inline components/esp_http_client/esp_http_client.c:636]: `config->client_key_password & config->client_key_password_len > 0`

Could be a single if statement.

File: components/esp_http_client/include/esp_http_client.h [hunk 2]
```diff
@@ -113,6 +113,8 @@ typedef struct {
     size_t                      client_cert_len;     /*!< Length of the buffer pointed to by client_cert_pem. May be 0 for null-terminated pem */
     const char                  *client_key_pem;     /*!< SSL client key, PEM format as string, if the server requires to verify client */
     size_t                      client_key_len;      /*!< Length of the buffer pointed to by client_key_pem. May be 0 for null-terminated pem */
+    const char                  *client_key_password;      /*!< Client key decryption password string */
+    int                         client_key_password_len;   /*!< String length of the password pointed to by clientkey_password */
```
Comment (AshUK) [inline components/esp_http_client/include/esp_http_client.h:117]: size_t

**Comment by AdityaHPatwardhan** [comment]
@AshUK Thank for the review. @joncmaloney can you please update the changes. I think that should be enough to merge this PR.

**Comment by joncmaloney** [comment]
@AshUK Thanks for the feedback. I've updated the changes suggested and rebased to espressif/esp-idf:master

**Comment by AxelLin** [comment]
@AdityaHPatwardhan 
Will this PR apply to v4.3? (It's a trivial change, the low level API already supports client key password)

**Comment by AdityaHPatwardhan** [comment]
Hi @AxelLin, so if it is required  we may apply this change to `v4.3` as well.

**Pull request closed without merging**